libnmstate/nm/nmclient.py

```diff
@@ -220,3 +234,53 @@ def _register_first_action(self):
     def _execute_action_once(self, _):
         self.execute_next_action()
         return False
+
+
+class _NmClient:
+    def __init__(self):
+        self.client = NM.Client.new(None)
+        logging.debug("NM.Client created")
+        self._mainloop = None
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, _exc_type, _exc_value, _traceback):
+        pass
+
+    def __del__(self):
+        try:
+            ref_count = sys.getrefcount(self.client)
+            if ref_count != 2:
+                raise error.NmstateInternalError(
+                    "_NmClient.__del__() with self.client holding "
+                    f"unexpected(not 2) reference count {ref_count}"
+                )
+            if NM.Client.get_version(self.client) >= StrictVersion("1.21.0"):
+                self._nmclient_cleanup()
+            else:
+                self._nmclient_cleanup_old()
+        except Exception:
+            pass
+
+    def _nmclient_cleanup_old(self):
+        self.client = None
+
+    def _nmclient_cleanup(self):
+        if self.client:
```

Comment:
Who may initialize this variable? If no one can/should, lets make it a property and avoid the condition here.